src/mvba/vcbc/mod.rs

```diff
@@ -19,12 +13,19 @@ use std::rc::Rc;
 
 pub(crate) const MODULE_NAME: &str = "vcbc";
 
+#[derive(Debug)]
+enum State {
+    Propose,
+    Echo,
+    Deliver,
+}
+
 // VCBC is a verifiably authenticatedly c-broadcast protocol.
 // Each party $P_i$ c-broadcasts the value that it proposes to all other parties
 // using verifiable authenticated consistent broadcast.
 pub(crate) struct Vcbc {
     ctx: context::Context,
-    state: Box<dyn State>,
```

Comment:
There are two way to implement state design pattern in Rust, using `Box<dyn state>` or `enum`. I prefer the first, If you don't mind let's do in this way. The actual algorithm is the same.

Reply:
We don't use traits in the code base when we can avoid it. In this case we can avoid it so please use them. It'll make the implementation fit into the rest of the codebase.

e.g. it took me some time to figure out what was happening with the trait based State, whereas this enum style will be familiar to anyone on the team.